packages/backend/src/connector/core/middleware/account.ts

```diff
@@ -34,6 +36,27 @@ export function createAccountMiddleware(serverAddress: string): ILPMiddleware {
           ) {
             throw new Errors.UnreachableError('destination account is disabled')
           }
+
+          // Create the tigerbeetle account if not exists.
+          // The incoming payment state will be PENDING until payments are received.
+          if (incomingPayment.state === IncomingPaymentState.Pending) {
+            try {
+              await ctx.services.accounting.createLiquidityAccount(
+                incomingAccount
+              )
+            } catch (err) {
+              // Don't complain if liquidity account already exists.
+              if (
+                err instanceof CreateAccountError &&
+                err.code === CreateAccountErrorCode.exists
+              ) {
+                // Do nothing.
+              } else {
+                throw err
+              }
+            }
+          }
```

Comment:
I'm not positive, but I think there is a race condition here. Assuming I'm reading things correctly, `incomingPayment.state` transitions from `Pending` to `Processing` _after_ the transfer[s] of money are committed. That means that this sequence is possible:

1. `getAccountByDestinationAddress` finds a pending `incomingPayment` and creates its liquidity account.
2. Commit `createTransfer` begins to commit, crediting the liquidity account.
3. (For unrelated reasons) Rafiki crashes before `destinationAccount.onCredit` can update the `state` to `Processing`.
4. Rafiki restarts and tries to create the account again — but it gets an `exists_with_different_{debits,credits}_{pending,posted}` error.

So I think those variants should be ignored here, like it ignores `exists`.

Reply:
Do you think `createLiquidityAccount` should catch errors and roll up all `exists*` errors as a single returnable error, a la `createTransfers`?
https://github.com/interledger/rafiki/blob/main/packages/backend/src/accounting/transfers.ts#L80

Reply:
Went with `areAllAccountExistsErrors([err.code])` that is used by other parts of the code from when tigerbeetle was updated